### PR TITLE
release-check-list: add pointer to operatorhub doc

### PR DESCRIPTION
**Pull request by wainersm**
On last release I created a document on CoCo's operator explaining how the bundle can be updated to the Operator Hub. Updated this release check-list to link to that document.

### CODE REVIEW

**Review by fitzthum — APPROVED** [review]
Thanks

**Review by fitzthum — APPROVED** [review]

**Comment by GabyCT** [comment]
lgtm

**Comment by fitzthum** [comment]
@confidential-containers/confidential-containers-maintainers can we get another review here

**Review by dcmiddle — APPROVED** [review]

**Pull request merged**
